storage/onpremise/mongodb/variables.tf

```
@@ -40,6 +40,37 @@ variable "mongodb" {
   })
 }
 
+variable "mongodb_resources" {
+  description = "CPU and Memory limits and requests for MongoDB"
+  type = object({
+    limits = optional(object({
+      cpu    = optional(string, "750m")
+      memory = optional(string, "768Mi")
+      }), {
+      cpu    = "750m"
+      memory = "768Mi"
+    })
+    requests = optional(object({
+      cpu    = optional(string, "500m")
+      memory = optional(string, "512Mi")
+      }), {
+      cpu    = "500m"
+      memory = "512Mi"
+    })
+  })
+  default = {
+    limits = {
+      cpu    = "750m"
+      memory = "768Mi"
+    }
+    requests = {
+      cpu    = "500m"
+      memory = "512Mi"
+    }
+  }
+}
```

You don't have to repeat the default values:
```suggestion
variable "mongodb_resources" {
  description = "CPU and Memory limits and requests for MongoDB"
  type = object({
    limits = optional(object({
      cpu    = optional(string, "750m")
      memory = optional(string, "768Mi")
      }), {})
    requests = optional(object({
      cpu    = optional(string, "500m")
      memory = optional(string, "512Mi")
      }), {})
  })
  default = {}
}
```